server/parsec/components/memory/shamir.py

```diff
@@ -0,0 +1,89 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+from parsec._parsec import DeviceID, OrganizationID, UserID, VerifyKey, authenticated_cmds
+from parsec.components.events import EventBus
+from parsec.components.memory.datamodel import (
+    MemoryDatamodel,
+    MemoryShamirSetup,
+)
+from parsec.components.shamir import (
+    BaseShamirComponent,
+    ShamirAddRecoverySetupBadOutcome,
+    VerifyCertificatesBadOutcome,
+    verify_certificates,
+)
+
+
+class MemoryShamirComponent(BaseShamirComponent):
+    def __init__(self, data: MemoryDatamodel, event_bus: EventBus) -> None:
+        super().__init__()
+        self._data = data
+        self._event_bus = event_bus
+
+    async def remove_recovery_setup(
+        self,
+        organization_id: OrganizationID,
+        author: UserID,
+    ) -> None:
+        # TODO update after https://github.com/Scille/parsec-cloud/issues/7364
+        self._data.organizations[organization_id].shamir_setup.pop(author)
```

Comment:
This method doesn't perform any of the expected checks.

Maybe merge it with `add_recovery_setup` (and rename to `shamir_recovery_setup`) to factorize the checks.

You could also duplicate the checks, they're already duplicated all around the code base anyway.